backend/jottacloud/jottacloud.go

```diff
@@ -748,6 +759,59 @@ func (o *Object) Open(options ...fs.OpenOption) (in io.ReadCloser, err error) {
 //
 // The new object may have been created if an error is returned
 func (o *Object) Update(in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) (err error) {
+	md5String, err := src.Hash(hash.MD5)
+	if err != nil || md5String == "" {
+		// if the source can't provide a MD5 hash we are screwed and need to calculate it ourselfs
```

Comment:
This little bit of code is now complicated enough that I'd factor it out into a function

```go
func readMD5(in io.Reader, size, threshold int64) (string, io.Reader, error) {}
```

and write a unit test that exercises all the paths in it.  The integration tests will only test the < 10M path.

Reply:
Awesome, thanks